crates/hotshot/src/tasks/mod.rs

```diff
@@ -260,64 +179,29 @@ pub async fn add_consensus_task<TYPES: NodeType, I: NodeImplementation<TYPES>>(
         .quorum_network
         .inject_consensus_info(ConsensusIntentEvent::PollForLatestQuorumProposal)
         .await;
-
-    // Poll (forever) for the latest view sync certificate
     consensus_state
         .quorum_network
         .inject_consensus_info(ConsensusIntentEvent::PollForLatestViewSyncCertificate)
         .await;
-
-    let filter = FilterEvent(Arc::new(consensus_event_filter));
-    let consensus_name = "Consensus Task";
-    let consensus_event_handler = HandleEvent(Arc::new(
-        move |event, mut state: ConsensusTaskState<TYPES, I, HotShotConsensusApi<TYPES, I>>| {
-            async move {
-                if let HotShotEvent::Shutdown = event {
-                    (Some(HotShotTaskCompleted::ShutDown), state)
-                } else {
-                    state.handle_event(event).await;
-                    (None, state)
-                }
-            }
-            .boxed()
-        },
-    ));
-    let consensus_task_builder = TaskBuilder::<
-        ConsensusTaskTypes<TYPES, I, HotShotConsensusApi<TYPES, I>>,
-    >::new(consensus_name.to_string())
-    .register_event_stream(event_stream.clone(), filter)
-    .await
-    .register_registry(&mut registry.clone())
-    .await
-    .register_state(consensus_state)
-    .register_event_handler(consensus_event_handler);
-    // impossible for unwrap to fail
-    // we *just* registered
-    let consensus_task_id = consensus_task_builder.get_task_id().unwrap();
-    let consensus_task = ConsensusTaskTypes::build(consensus_task_builder).launch();
-
-    task_runner.add_task(
-        consensus_task_id,
-        consensus_name.to_string(),
-        consensus_task,
-    )
+    consensus_state
+    // let task = Task::new(tx, rx, task_reg.clone(), consensus_state);
+    // task_reg.run_task(task).await;
```

Comment:
Why don't we run the consensus task at the end of the function as we do for other tasks?

Reply:
I should clean this up, I think it was because we needed the consensus state for the task tests, but the pattern should match for all tasks.  I'll try to fix that